src/NewsExtraFieldDisplay.php

```diff
@@ -138,42 +138,45 @@ public function formAlter(array &$form, FormStateInterface $form_state, $form_id
     ) {
 
       $form_object = $form_state->getFormObject();
-      if ($form_object instanceof NodeForm) {
-        $node = $form_object->getEntity();
-        if (empty($this->moderationInformation) || !$this->moderationInformation->isModeratedEntity($node)) {
-          $visible = [
-            ":input[name='status[value]']" => [
-              'checked' => TRUE,
-            ],
-          ];
-        }
-        else {
-          $workflow = $this->moderationInformation->getWorkflowForEntity($node);
-          $type_plugin = $workflow->getTypePlugin();
-          $transitions = $type_plugin->getTransitions();
-          $published = [];
-          foreach ($transitions as $transition) {
-            $state = $transition->to();
-            if ($state->isPublishedState()) {
-              $published[] = [":input[name='moderation_state[0][state]']" => ['value' => $state->id()]];
-              $published[] = 'or';
-            }
-          }
-          array_pop($published);
-          $visible = [$published];
-        }
 
-        $form['localgov_news_newsroom_promote'] = [
-          '#title' => $this->t('Promote on newsroom'),
-          '#type' => 'checkbox',
-          '#description' => $this->t("Add to promoted news in the newsroom. If there is already the maximum number of promoted news items the last will be removed to make space."),
-          '#default_value' => self::articlePromotedStatus($form_object),
-          '#states' => [
-            'visible' => $visible,
+      // Must be a content entity form on a node.
+      assert($form_object instanceof ContentEntityFormInterface);
+      $node = $form_object->getEntity();
+      assert($node instanceof NodeInterface);
```

Comment:
Fixed a typo here
`assert($node instanceof $node);` should have been `assert($node instanceof NodeInterface);`